src/object/cli_obj.c

```diff
@@ -2342,11 +2546,29 @@ obj_comp_cb(tse_task_t *task, void *data)
 			D_ASSERT(d_list_empty(head));
 		}
 		obj_bulk_fini(obj_auxi);
-		obj_reasb_req_fini(obj_auxi);
-		/* zero it as user might reuse/resched the task, for example
-		 * the usage in dac_array_set_size().
-		 */
-		memset(obj_auxi, 0, sizeof(*obj_auxi));
+
+		fail_info = obj_auxi->reasb_req.orr_fail;
+		if (task->dt_result == 0 && fail_info != NULL) {
```

Comment:
I am a bit confused here. I thought degraded fetch is triggered if fetching a data shard failed.  so it means it will go through the similar process as replication case. 

refresh the pool_map. fetch other data_stripe + parity, reconstruct data. 

so why you put these in !io_retry block?

Reply:
In this patch it keeps the normal retryable failure same as before, such as RPC timeout etc.

Here it is for the case that 1) if some recxs need to be recovered, then create a set of recovery fetch tasks to get back needed stripe data by obj_ec_recov_cb, and 2) when the recovery fetch tasks finished it need to re-construct the missed data and copy to original user buffer, by obj_ec_recov_data.

Reply:
I think retry means it will try to retrieve the data for this shard in an "alternative" way. 
For replication data, it will try to get from different replicas, probably not implemented yet, but we should.
For EC, it will reconstruct data right away(not reread the same shard), instead of waiting the pool map change, which might take long time.

Reply:
I thought a while, I tend to think that purely RPC timeout or other temporarily network failure probably should not trigger EC degraded fetch, because 1) degraded fetch is expensive actually, and 2) try to make the retry simpler and same as before.

EC degraded fetch will be triggered when 1) target shard not existing or in rebuilding; and 2) csum error.
In this patch only triggered by case 1).
CSUM error case not added now, and I am not sure if need to trigger degraded fetch immediately for csum error (or after 2 or 3 times normal retry).